iothub/service/tests/IotHubServiceClientTests.cs

```diff
@@ -0,0 +1,122 @@
+            serviceClient.MessageFeedback.Should().NotBeNull();
+            serviceClient.FileUploadNotifications.Should().NotBeNull();
+            serviceClient.Messages.Should().NotBeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceClient_CreateSubClientsWithAAD()
+        {
+            // arrange
+            string hostName = "acme.azure-devices.net";
+            var tokenCrediential = new TestTokenCredential(new DateTimeOffset(DateTime.Now + TimeSpan.FromHours(1)));
+
+            // act
+            using var serviceClient = new IotHubServiceClient(hostName, tokenCrediential);
+
+            // assert
+            serviceClient.Devices.Should().NotBeNull();
+            serviceClient.Modules.Should().NotBeNull();
+            serviceClient.Configurations.Should().NotBeNull();
+            serviceClient.DirectMethods.Should().NotBeNull();
+            serviceClient.Query.Should().NotBeNull();
+            serviceClient.ScheduledJobs.Should().NotBeNull();
+            serviceClient.DigitalTwins.Should().NotBeNull();
+            serviceClient.Twins.Should().NotBeNull();
+            serviceClient.MessageFeedback.Should().NotBeNull();
+            serviceClient.FileUploadNotifications.Should().NotBeNull();
+            serviceClient.Messages.Should().NotBeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceClient_CreateSubClientsWithSasToken()
+        {
+            // arrange
+            string hostName = "acme.azure-devices.net";
+            var sasCredential = new AzureSasCredential("test");
+
+            // act
+            using var serviceClient = new IotHubServiceClient(hostName, sasCredential);
+
+            // assert
+            serviceClient.Devices.Should().NotBeNull();
+            serviceClient.Modules.Should().NotBeNull();
+            serviceClient.Configurations.Should().NotBeNull();
+            serviceClient.DirectMethods.Should().NotBeNull();
+            serviceClient.Query.Should().NotBeNull();
+            serviceClient.ScheduledJobs.Should().NotBeNull();
+            serviceClient.DigitalTwins.Should().NotBeNull();
+            serviceClient.Twins.Should().NotBeNull();
+            serviceClient.MessageFeedback.Should().NotBeNull();
+            serviceClient.FileUploadNotifications.Should().NotBeNull();
+            serviceClient.Messages.Should().NotBeNull();
+        }
+
+        [TestMethod]
+        public void IotHubServiceClient_NullParamters_Throws()
+        {
+            // arrange
+            string hostName = null;
+            var sasCredential = new AzureSasCredential("test");
+
+            try
+            {
+                // act
+                using var serviceClient = new IotHubServiceClient(hostName, sasCredential);
+            }
+            catch (Exception ex)
+            {
+                // assert
+                ex.Should().BeOfType<ArgumentNullException>();
+            }
```

Comment:
Same here, use FluentAssertions instead of try/catch.

If it doesn't throw, then the test wouldn't fail as expected.